code/go/0chain.net/smartcontract/faucetsc/update_test.go

```diff
@@ -0,0 +1,227 @@
+package faucetsc
+
+import (
+	"testing"
+	"time"
+
+	configpkg "0chain.net/chaincore/config"
+	"0chain.net/core/common"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func assertErrMsg(t *testing.T, err error, msg string) {
+	t.Helper()
+
+	if msg == "" {
+		assert.Nil(t, err)
+		return
+	}
+
+	if assert.NotNil(t, err) {
+		assert.Equal(t, msg, err.Error())
+	}
+}
+
+func getGlobalNodeTest() (gn *GlobalNode) {
+	const pfx = "smart_contracts.faucetsc."
+
+	configpkg.SmartContractConfig.Set(pfx+"pour_amount", 100)
+	configpkg.SmartContractConfig.Set(pfx+"max_pour_amount", 1000)
+	configpkg.SmartContractConfig.Set(pfx+"periodic_limit", 5000)
+	configpkg.SmartContractConfig.Set(pfx+"global_limit", 1000000)
+	configpkg.SmartContractConfig.Set(pfx+"individual_reset", 24*time.Hour)
+	configpkg.SmartContractConfig.Set(pfx+"global_reset", 240*time.Hour)
+
+	return &GlobalNode{
+		ADDRESS,
+		100, 1000, 5000, 1000000,
+		24 * time.Hour, 240 * time.Hour,
+		0, common.ToTime(common.Timestamp(0)),
+	}
+}
+
+func Test_getConfig(t *testing.T) {
+	var (
+		fsc        = newTestFaucetSC()
+		balances   = newTestBalances()
+		tp         = common.Timestamp(0)
+		tx         = newTransaction(owner, fsc.ID, 0, tp)
+		configured = getGlobalNodeTest()
+		gn         = fsc.getGlobalVariables(tx, balances)
+	)
+	assert.EqualValues(t, configured, gn)
+}
+
+func TestFaucetSmartContractUpdate(t *testing.T) {
+	var (
+		fsc        = newTestFaucetSC()
+		balances   = newTestBalances()
+		tp         = common.Timestamp(0)
+		tx         = newTransaction(owner, fsc.ID, 0, tp)
+		originalGn = getGlobalNodeTest()
+		gn         = fsc.getGlobalVariables(tx, balances)
+		lr         = &limitRequest{}
+		err        error
+	)
+
+	balances.txn = tx
+
+	// 1. Malformed limit request
+	t.Run("malformed request", func(t *testing.T) {
+		_, err = fsc.updateLimits(tx, []byte("} malformed {"), balances, gn)
```

Comment:
All these tests have a similar format, calling`_fsc.updateLimits` with various setup condtions. 
For consistancy can now we adopt the [table driven subtest style](https://blog.golang.org/subtests#TOC_4.) of @MurashovVen's unit tests.